rust/src/database/verkle/compute_commitment.rs

```diff
@@ -50,11 +50,47 @@ pub fn compute_leaf_node_commitment(
     let c1 = Commitment::new(&values[0]);
     let c2 = Commitment::new(&values[1]);
 
-    let combined = vec![
+    let combined = [
         Scalar::from(1),
         Scalar::from_le_bytes(stem),
         c1.to_scalar(),
         c2.to_scalar(),
     ];
     Commitment::new(&combined)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::database::verkle::test_utils::FromIndexValues;
+
+    #[test]
+    fn compute_leaf_node_commitment_produces_expected_values() {
+        {
+            let values = [Value::default(); 256];
+            let used_bits = [0; 256 / 8];
+            let stem = [0u8; 31];
+            let commitment = compute_leaf_node_commitment(&values, &used_bits, &stem);
+            let expected = Commitment::new(&[Scalar::from(1)]);
```

Comment:
I'm probably overlooking something, but why is the commitment of empty values the same as the commitment of 1?

Reply:
It just works out that way because C1, C2 and stem all end up being the zero commitment. The final commitment is then `commitment(1, stem, C1, C2)`.

Reply:
Ah that magic `1` 😆